backend/src/roles/roles.service.ts

```diff
@@ -1,53 +1,84 @@
 import { Injectable } from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
+
+export class RoleDetails {
+  id!: string;
+  name!: string;
+  description?: string;
+  permissions?: string[];
+}
```

Comment:
No need for `RoleDetails`

Prisma provides the the class itself. 

import { Role } from '@prisma/client';